### PR TITLE
Add the link to Deployment API

### PR DESCRIPTION
**Pull request by khuyentran1401**
When reading the section about creating a deployment from a Python object, I found it difficult to know the possible parameters for the `Deployment` object. So I think it will be helpful to link to the API for the `Deployment` object.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

File: docs/concepts/deployments.md [hunk 1]
```diff
@@ -405,6 +405,8 @@ deployment.load() # loads server-side settings
 
 Once the existing deployment settings are loaded, you may update them as needed by changing deployment properties.
 
+View all of the parameters for the `Deployment` object [here](https://docs.prefect.io/api-ref/prefect/deployments/).
```
Comment (tpdorsey) [inline docs/concepts/deployments.md:408]:
```suggestion
View all of the parameters for the `Deployment` object in the [Python API documentation](https://docs.prefect.io/api-ref/prefect/deployments/).
```

**Review by tpdorsey — COMMENTED** [review]
Minor suggested edit

**Comment by tpdorsey** [comment]
@serinamarie this PR also includes adding an item to the recipes catalog. Can you weigh in on this?

File: docs/recipes/catalog/github-docker.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+recipeName: Deployment with GitHub and Docker
```
Comment (serinamarie) [inline docs/recipes/catalog/github-docker.yaml:1]:
```suggestion
recipeName: Flow Deployment with GitHub Storage and Docker Infrastructure
```

**Review by serinamarie — CHANGES_REQUESTED** [review]
Love this idea and think it would be great to have a list of all available parameters at the ready like this for all our primitives, rather than a user having to go to the search bar again.

**Review by tpdorsey — APPROVED** [review]

**Pull request merged**